torch/prototypes/nestedtensor/__init__.py

```diff
@@ -0,0 +1,64 @@
+import torch
+from . import nested
+import torch.prototypes.nestedtensor.codegen as codegen
+
+def is_available():
+    return hasattr(torch._C, "_nestedtensor_init")
+
+if is_available() and not torch._C._nestedtensor_init():
+    raise RuntimeError("Failed to initialize PyTorch nestedtensor support")
+
+import torch.prototypes.nestedtensor as nestedtensor
+
+NestedTensor = nested.NestedTensor
+
+def _create_out(input1, out, dtype=None):
+    if out is None:
+        if dtype is None:
+            dtype = input1.dtype
+        out_tensors = []
+        for tensor in input1.tensors:
+            out_tensors.append(torch.empty_like(tensor, dtype=dtype))
+        out = NestedTensor(out_tensors)
+    assert len(out) == len(input1)
+    return out
+
+
+def _unary(func_name, func, input1, out=None):
+    out = _create_out(input1, out)
+    for i in range(len(out)):
+        # NOTE: We are disabling broadcasting for now
+        assert out.tensors[i].size() == input1.tensors[i].size()
+    list_func = getattr(nestedtensor, func_name)
+    list_func(input1.tensors, out.tensors)
+    return out
+
+
+# The contract is that func only works with torch.Tensor
+def _binary(func_name, func, input1, input2, out=None):
+    out = _create_out(input1, out)
+    assert len(input1) == len(input2)
+    for i in range(len(out)):
+        # NOTE: We are disabling broadcasting for now
+        assert out.tensors[i].size() == input1.tensors[i].size()
+        assert input2.tensors[i].size() == input1.tensors[i].size()
+    list_func = getattr(nestedtensor, func_name)
+    list_func(input1.tensors, input2.tensors, out.tensors)
+    return out
+
+
+def _comparison(func_name, func, input1, input2, out=None):
```

Comment:
This is almost the same thing as binary. A `_binary(..., out_dtype=None)` function could cover both cases

Reply:
unfortunately that's not the signature of torch.lt etc. - the wrapper passes the arguments to those functions before prepending the function name and a reference to the original function.